cmake/CompilerOptions.cmake

```diff
@@ -18,20 +21,60 @@ if(${ARBDEV_COLOR})
 endif()
 
 # A library to collect compiler-specific linking adjustments.
-
 add_library(arbor-compiler-compat INTERFACE)
-# TODO Remove when upgrading GCC.
-if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.1)
-    target_link_libraries(arbor-compiler-compat INTERFACE stdc++fs)
-  endif()
-endif()
-# TODO Remove when upgrading Clang
-if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)
-    target_link_libraries(arbor-compiler-compat INTERFACE stdc++fs)
+
+cmake_push_check_state()
+# Check how to use std::filesystem
+string(CONFIGURE [[
+  #include <cstdlib>
```

Comment:
For `printf` this should be `#include <cstdio>`, and `std::printf` below.

Reply:
True.